android/fakedapp/src/main/java/com/solana/mobilewalletadapter/fakedapp/MainViewModel.kt

```diff
@@ -321,351 +394,99 @@ class MainViewModel(application: Application) : AndroidViewModel(application) {
-    private fun doSignAndSendTransactions(
-        client: MobileWalletAdapterClient,
-        transactions: Array<ByteArray>,
-        minContextSlot: Int? = null
-    ): Array<ByteArray>? {
-        var signatures: Array<ByteArray>? = null
+    private suspend fun doDeauthorize(
+        client: MobileWalletAdapterUseCase.Client,
+        currentAuthToken: String
+    ) {
         try {
-            val result = client.signAndSendTransactions(transactions, minContextSlot).get()
-            Log.d(TAG, "Signatures: ${result.signatures.contentToString()}")
-            signatures = result.signatures
-        } catch (e: ExecutionException) {
-            when (val cause = e.cause) {
-                is IOException ->
-                    Log.e(TAG, "IO error while sending sign_and_send_transactions", cause)
-                is TimeoutException ->
-                    Log.e(TAG, "Timed out while waiting for sign_and_send_transactions result", cause)
-                is MobileWalletAdapterClient.InvalidPayloadsException ->
-                    Log.e(TAG, "Transaction payloads invalid", cause)
-                is MobileWalletAdapterClient.NotSubmittedException -> {
-                    Log.e(TAG, "Not all transactions were submitted", cause)
-                }
-                is JsonRpc20Client.JsonRpc20RemoteException ->
-                    when (cause.code) {
-                        ProtocolContract.ERROR_AUTHORIZATION_FAILED -> Log.e(TAG, "Authorization invalid, authorization or reauthorization required", cause)
-                        ProtocolContract.ERROR_NOT_SIGNED -> Log.e(TAG, "User did not authorize signing", cause)
-                        ProtocolContract.ERROR_TOO_MANY_PAYLOADS -> Log.e(TAG, "Too many payloads to sign", cause)
-                        else -> Log.e(TAG, "Remote exception for sign_and_send_transactions", cause)
-                    }
-                is JsonRpc20Client.JsonRpc20Exception ->
-                    Log.e(TAG, "JSON-RPC client exception for sign_and_send_transactions", cause)
-                else -> throw e
+            client.deauthorize(currentAuthToken)
+        } finally {
+            _uiState.update {
+                it.copy(
+                    authToken = null,
+                    publicKey = null,
+                    accountLabel = null,
+                    walletUriBase = null
+                )
             }
-        } catch (e: CancellationException) {
-            Log.e(TAG, "sign_and_send_transactions request was cancelled", e)
-        } catch (e: InterruptedException) {
-            Log.e(TAG, "sign_and_send_transactions request was interrupted", e)
         }
-
-        return signatures
     }
 
-    private suspend fun <T> localAssociateAndExecute(
-        sender: StartActivityForResultSender,
+    private suspend fun <T> doLocalAssociateAndExecute(
+        intentLauncher: ActivityResultLauncher<MobileWalletAdapterUseCase.StartMobileWalletAdapterActivity.CreateParams>,
```

Comment:
This feels unnecessarily long and hard to read. You can put the objects beside each other.

Reply:
Imported `StartMobileWalletAdapterActivity` to reduce qualified class name string length